### PR TITLE
Raise with nice error if incoming-changes hook returns bad value (fixes #425)

### LINKED ISSUE #425: Warn instead of crash when incoming-changes hook returns nothing
Currently when configured incoming-changes hook returns nothing, it crashes:
```
Unhandled promise rejection TypeError: t is undefined
Trace de la pile :
[316]</Collection</<.value@http://npmcdn.com/kinto@2.0.4/dist/kinto.min.js:952:8745
[316]</Collection</<.value/<@http://npmcdn.com/kinto@2.0.4/dist/kinto.min.js:952:11181
[201]</notify/</s@http://npmcdn.com/kinto@2.0.4/dist/kinto.min.js:605:1730
[201]</notify/<@http://npmcdn.com/kinto@2.0.4/dist/kinto.min.js:605:1869
[66]</module.exports/s@http://npmcdn.com/kinto@2.0.4/dist/kinto.min.js:200:343
```

### PR DESCRIPTION
**Pull request by leplatrem**
@n1k0 r?

### CODE REVIEW

File: src/collection.js [hunk 1]
```diff
@@ -787,7 +787,13 @@ export default class Collection {
       return Promise.resolve(payload);
     }
     return waterfall(this.hooks[hookName].map(hook => {
-      return record => hook(payload, this);
+      return record => {
+        const result = hook(payload, this);
+        if (!result || !result.hasOwnProperty("changes")) {
+          throw new Error(`Invalid return value for hook: ${result} has no 'changes' property`);
```
Comment (n1k0) [inline src/collection.js]: This is likely to render as "Invalid return value for hook: [Object object] has no 'changes' property"

**Comment by leplatrem** [comment]
@n1k0 did not understand your last comment.

**Comment by leplatrem** [comment]
```js
>> const v = 42;
undefined
>> v.hasOwnProperty("changes")
false
```

**Comment by n1k0** [comment]
r+ with a green build :)

**Pull request merged**